fluid/PaddleRec/ctr/reader.py

```diff
@@ -0,0 +1,74 @@
+class Dataset:
+    def _reader_creator(self, file_list, is_infer):
+        def reader():
+            for file in file_list:
+                with open(file, 'r') as f:
+                    for line in f:
+                        features = line.rstrip('\n').split('\t')
+                        dense_feature = map(float, features[0].split(','))
+                        sparse_feature = map(lambda x: [int(x)], features[1].split(','))
+                        if not is_infer:
+                            label = [float(features[2])]
+                            yield [dense_feature
+                                   ] + sparse_feature + [label]
+                        else:
+                            yield [dense_feature] + sparse_feature
+
+        return reader
+
+    def train(self, file_list):
+        return self._reader_creator(file_list, False)
+
+    def test(self, file_list):
+        return self._reader_creator(file_list, False)
+
+    def infer(self, file_list):
+        return self._reader_creator(file_list, True)
+
+class CriteoDataset:
+    def __init__(self, sparse_feature_dim):
+        self.cont_min_ = [0, -3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        self.cont_max_ = [20, 600, 100, 50, 64000, 500, 100, 50, 500, 10, 10, 10, 50]
+        self.cont_diff_ = [20, 603, 100, 50, 64000, 500, 100, 50, 500, 10, 10, 10, 50]
+        self.hash_dim_ = sparse_feature_dim
+        self.train_idx_ = 41256555
+        self.continuous_range_ = range(1, 14)
+        self.categorical_range_ = range(14, 40)
+
+    def _reader_creator(self, file_list, is_train, trainer_id):
+        def reader():
+            for file in file_list:
+                with open(file, 'r') as f:
+                    line_idx = 0
+                    for line in f:
+                        line_idx += 1
+                        if is_train and line_idx > self.train_idx_:
+                            continue
+                        elif not is_train and line_idx <= self.train_idx_:
+                            continue
+                        if trainer_id > 0 and line_idx % trainer_id != 0:
```

Comment:
This place should use 
```python
line_idx % trainer_num != trainer_idx
```
for example,  `6 % 3` and `6 % 2` is both 0.

Reply:
will fix, thx